src/classes/GeoIP.mjs

```diff
@@ -23,4 +23,20 @@ export default class GeoIP {
   static lookup (ip) {
     return { ...cityLookup.get(ip), ...asnLookup.get(ip) }
   }
+
+  /**
+   * Get the GeoIP information about an IP address as a string
+   * @param {string} ip the ip address to look up
+   * @returns {string} geoip information string
+   */
+  static locationString (ip) {
+    const geoip = GeoIP.lookup(ip)
+    if (geoip.city && geoip.country) {
+      return `${geoip.city.names.en}, ${geoip.postal.code} ${geoip.country.names.en}`
+    }
+    if (geoip.country) {
+      return `${geoip.city.names.en}, ${geoip.postal.code} ${geoip.country.names.en}`
```

Comment:
this block will always cause an error when its executed since we know from the previous block that city does not exist.

```suggestion
      return geoip.country.names.en
```

Reply:
Alternatively you could provide a city placeholder like:

```suggestion
      return `Unknown City, ${geoip.country.names.en}`
```